File: examples/sdk/js/replayMembrane.ts
import { readFileSync } from 'node:fs';
import { resolve } from 'node:path';
import { replayMembrane } from '../../../packages/parva-local-kernel/src/index';

const path = process.argv[2] ?? 'examples/external/proofpacks/civil-conversion.proofpack.json';
const artifact = JSON.parse(readFileSync(resolve(path), 'utf8'));
const membrane = artifact.membrane ?? artifact;
const result = await replayMembrane(membrane, []);

console.log(JSON.stringify({ ...result, operation: membrane.canonical_query?.operation }, null, 2));
process.exit(result.verified ? 0 : 1);

File: examples/sdk/python/replay_membrane.py
from __future__ import annotations

import json
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[3]
sys.path[:0] = [str(ROOT), str(ROOT / "backend"), str(ROOT / "packages/parva-python")]

from app.membranes.verifier import verify_membrane  # noqa: E402


def main(path: str = "examples/external/proofpacks/civil-conversion.proofpack.json") -> int:
    artifact = json.loads((ROOT / path).read_text(encoding="utf-8"))
    membrane = artifact.get("membrane", artifact)
    ok, reason = verify_membrane(membrane)
    print(json.dumps({"verified": ok, "reason": reason, "operation": membrane.get("canonical_query", {}).get("operation")}, indent=2))
    return 0 if ok else 1


if __name__ == "__main__":
    raise SystemExit(main(sys.argv[1] if len(sys.argv) > 1 else "examples/external/proofpacks/civil-conversion.proofpack.json"))
